packages/common/src/ContractInteractor.ts

```diff
@@ -1129,3 +1129,22 @@ export function getRawTxOptions (chainId: number, networkId: number, chain?: str
       }, 'london')
   }
 }
+
+export async function getChainId (provider: Web3ProviderBaseInterface): Promise<number> {
```

Comment:
we're doing it too many times: better create `async sendAsync(provider,method, ...params)`, which performs callback handling and use that.